monsterCatalog.py

```diff
@@ -188,180 +190,143 @@ def monsters():
-        oauth_flow.redirect_uri = 'postmessage'
-        credentials = oauth_flow.step2_exchange(code)
-    except FlowExchangeError:
-        response = flask.make_response(
-            json.dumps('Failed to upgrade the authorization code.'), 401)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    # Check that the access token is valid.
-    access_token = credentials.access_token
-    url = ('https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=%s'
-           % access_token)
-    h = httplib2.Http()
-    result = json.loads(h.request(url, 'GET')[1])
-    # If there was an error in the access token info, abort.
-    if result.get('error') is not None:
-        response = flask.make_response(json.dumps(result.get('error')), 500)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    # Verify that the access token is used for the intended user.
-    gplus_id = credentials.id_token['sub']
-    if result['user_id'] != gplus_id:
-        response = flask.make_response(
-            json.dumps("Token's user ID doesn't match given user ID."), 401)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    # Verify that the access token is valid for this app.
-    if result['issued_to'] != CLIENT_ID:
-        response = flask.make_response(
-            json.dumps("Token's client ID does not match app's."), 401)
-        print "Token's client ID does not match app's."
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    stored_credentials = flask.session.get('credentials')
-    stored_gplus_id = flask.session.get('gplus_id')
-    if stored_credentials is not None and gplus_id == stored_gplus_id:
-        response = flask.make_response(json.dumps('Current user is already connected.'), 200)
-        response.headers['Content-Type'] = 'application/json'
-        return response
-
-    # Store the access token in the session for later use.
-    flask.session['credentials'] = credentials.access_token
-    flask.session['gplus_id'] = gplus_id
-
-    # Get user info
-    userinfo_url = "https://www.googleapis.com/oauth2/v1/userinfo"
-    params = {'access_token': credentials.access_token, 'alt': 'json'}
-    answer = requests.get(userinfo_url, params=params)
-
-    data = answer.json()
-
-    flask.session['username'] = data['name']
-    flask.session['picture'] = data['picture']
-    flask.session['email'] = data['email']
-
-    output = ''
-    output += '<h1>Welcome, '
-    output += flask.session['username']
-    output += '!</h1>'
-    output += '<img src="'
-    output += flask.session['picture']
-    output += ' " style = "width: 300px; height: 300px;border-radius: 150px;-webkit-border-radius: 150px;-moz-border-radius: 150px;"> '
-    flask.flash("you are now logged in as %s" % flask.session['username'])
-    print "done!"
-    return output
+
+def set_state():
```

Comment:
What is "state"?  Looks like it's a 32-char random string.  You might want to name it something relevant to what it's really for.  State tells me nothing.